### DISCUSSION TITLE
Can't build episode 189 code

### DISCUSSION THREAD

**Discussion by jharvey3** [pointfreeco/episode-code-samples#115] [Q&A] [answered]
I pulled the code for episode #189 (Tour of Parser=Printers: API Clients for Free), and had 2 problems I don't understand trying to build the Run scheme for the Server in Xcode 13.3.1.

The first is that configure.swift the compiler complains that import Vapor won't work because:
"Compiling for iOS 9.0, but module 'Vapor' has a minimum deployment target of iOS 13.0:"

I can fix this problem by changing platforms in the Package.swift file for the Server to: 
[
    .macOS(.v12),
    .iOS(.v13)
],

Which is annoying. But I guess understandable.  

The compiler also complains that this code in the configure function in configure.swift:
app.router = router
    .baseURL("http://127.0.0.1:8080")
    .eraseToAnyParserPrinter()

is wrong for the following reason: 

Cannot assign value of type 'AnyParserPrinter<URLRequestData, SiteRoute>' to type 'SiteRouterKey.Value' (aka 'AnyParserPrinter<URLRequestData, SiteRoute>')

It says that SiteRouterKey.Value is an AnyParserPrinter<URLRequestData, SiteRoute> but it doesn't allow an AnyParserPrinter<URLRequestData, SiteRoute> to be assigned to it?

This last problem is a mystery to me.  Any help would be appreciated.

This is on a MacBook Pro (13-inch, M1, 2020), Monterery 12.3.1.  Xcode 13.3.1

**Comment by jharvey3** [comment]
One note.  I rechecked out the code for the episode and the issue about Vapor having the wrong minimum deployment target is not reproducible.  This is after restarting Xcode.  The second problem remains.

**Comment by mbrandonw** [comment, marked as answer]
Thanks for the report! 

When we recorded the episode we used _URLRouting that is vended from swift-parsing, but when we released the episode we also released the official non-underscored URLRouting in the swift-url-routing repo. So this code sample was technically using two versions of the routing library at once, causing ambiguity. 

I have now (c489eb5) updated all _URLRouting to be just URLRouting and it should work.

**Reply by jharvey3** [reply]
Yes everything is fine now.  Thanks.
